server/reports/attendance/attendance-endpoint.js

```diff
@@ -1,155 +1,167 @@
+                classId,
+                classes.Classes[i].ClassDescription.Name,
+                classes.Classes[i].StartDateTime,
+                classes.Classes[i].MaxCapacity, 
+                classes.Classes[i].TotalBooked, 
+                numberAttended
+              ]); // pushes current class's data to attendanceReport
+            }
+          })
+          .catch((err) => {
+            console.log("ERROR! ", err);
+          });
+        allNumberAttendedPromises.push(numberAttendedPromise);
+      }
+
+      //resolves all, allNumberAttendedPromises then outputs attendance report to endpoint as CSV or JSON
+      return Promise.all(allNumberAttendedPromises)
+        .then(() => {
+          if (format === "csv") {
+            let parser = new J2C.Parser(attendanceReport.headers);
+            let dataForParser = attendanceReport.data.map((row) => {
+              let rowForParser = {};
+              for (let i = 0; i < attendanceReport.headers.length; i ++) {
+                rowForParser[attendanceReport.headers[i]] = row[i];
+              }
+              return rowForParser;
+            });
+            return parser.parse(formatJSON(attendanceReport.data, attendanceReport.headers));
+          }
+          return attendanceReport;
+        });
+    });
+}
+
+//gets attendance data from mindbody endpoints and outputs it as .csv or JSON to our endpoint
+//endpoint URL example
+//http://localhost:8080/reports/attendance?format=csv&startdate=01/01/2020&enddate=12/31/2020&minStudents=4
+export function attendanceRequestHandler(request, response) {
+  let format = request.query.format; //gets format value in URL query
+  let startdate = request.query.startdate; //gets startdate value in URL query
+  let enddate = request.query.enddate; //gets enddate value in URL query
+  let minStudents = Number(request.query.minStudents);
+
+  //error handling
+  //"throws error" if format isn't JSON or CSV
+  if (!(format === "json" || format === "csv")) {
+    response.send("Bad format parameter. Must be \"json\" or \"csv\"");
+    return;
+  }
+  //sets startdate to 2 weeks prior to current day if field is NULL
+  if (!startdate) {
+    let today = new Date();
+    today.setDate(today.getDate()-14);
+    let twoWeeksPrior = reformatDate(today);
+    startdate = twoWeeksPrior;
+  }
+  //sets enddate to current day if field is NULL
+  if (!enddate) {
+    let today = new Date();
+    enddate = reformatDate(today);
+  }
+  //sets minStudents to MINSTUDENTS_DEFAULT if field is NaN
+  if (Number.isNaN(minStudents)) {
+    minStudents = MINSTUDENTS_DEFAULT;
+  }
+
+  getAttendanceReport(format, startdate, enddate, minStudents)
+    .then((report) => {
+      if (format == "csv") {
+        let fileName = "AttendanceReport.csv";
+        //todo: we should choose a filename that includes helpful info like the date range
+        response.setHeader("Content-Disposition", "attachment ; filename=\"" + fileName + "\"");
```

Comment:
Be careful to not overwrite Jason's filename changes here

Reply:
Thanks for catching that. I'm changing it right now.